modules/shadertools/bundle.ts

```diff
@@ -0,0 +1,3 @@
+// Re-export core API so they don't get overwritten
+export * from '@luma.gl/core';
+export * from '@luma.gl/shadertools';
```

Comment:
- This is intentional in the sense that I keep shadertools independent of core (does not import core but works with code). 
- The price for this is a couple of small redefinitions.
- As shadertools is a string processing library and does not need the adapter, it makes a lot of sense to keep it independent to avoid dependency creep.

Reply:
Presumably this means that this bundle does not need to include core, as there should be no dependency. Also might be possible to remove the peer dependency?

Reply:
Not difficult to remove, but would like your take on how to approach it:

```
modules/shadertools/src/lib/wgsl/get-shader-layout-wgsl.ts:
  5: import {ShaderAttributeType, ShaderLayout, log} from '@luma.gl/core';

modules/shadertools/src/modules/lighting/pbr-material/pbr-material.ts:
  7: import type {NumberArray, Texture} from '@luma.gl/core';
```

Reply:
The WGSL is the trickier one. We'd need to duplicate the ShaderLayout type which is pretty involved. If we wanted to make the case for breaking out some common utils from core, it would be around these types and the gpu-type-utils folder in the lated modules/core/src

As an alternative, we could also regard WGSL parsing as the province of the webgpu module which has those types already/ the core dependency already.

Reply:
For the second, import NumericArray from math.gl/types and use a dummy type Texture=unknownl style type definition